presto-main/src/main/java/com/facebook/presto/sql/planner/SubqueryPlanner.java

```diff
@@ -132,10 +136,11 @@ public PlanBuilder handleSubqueries(PlanBuilder builder, Expression expression,
 
     private PlanBuilder handleSubqueries(PlanBuilder builder, Expression expression, Node node, boolean correlationAllowed, SqlPlannerContext context)
     {
-        builder = appendInPredicateApplyNodes(builder, collectInPredicateSubqueries(expression, node), correlationAllowed, node, context);
+        boolean antiJoinContext = extractAntiJoinContext(expression);
+        builder = appendInPredicateApplyNodes(builder, collectInPredicateSubqueries(expression, node), correlationAllowed, antiJoinContext, node, context);
         builder = appendScalarSubqueryApplyNodes(builder, collectScalarSubqueries(expression, node), correlationAllowed, context);
         builder = appendExistsSubqueryApplyNodes(builder, collectExistsSubqueries(expression, node), correlationAllowed, context);
```

Comment:
Do these two calls need to pass this antiJoinContext?

Reply:
Fair enough. The tradeoff is between code hygiene and consistency vs a small amount of additional processing. I will make this consistent